docs/data-management/definitions/fact-sql.md

```diff
@@ -4,16 +4,41 @@ sidebar_position: 4
-3. **Fact Values (optional)** - Optional values associated with the event (revenue amount, minutes watched, etc.).
-4. **Fact Properties (optional)** - Optional properties associated with the event (`purchase_type`, etc.). Note that properties that are 1:1 with subjects should be defined in the [Assignment](/data-management/definitions/assignment-sql) or [Entity Property](/data-management/definitions/property-sql) SQL definition.
+**Fact SQL Definitions** are short SQL snippets that tell Eppo how to use the data in your data warehouse. When you create [metrics](/data-management/metrics/) in Eppo, you specify one (or more) facts and an aggregation method. For instance, if you have a fact `revenue`, you could have a metric `Total Revenue` that computes the sum of revenue facts.
 
+Eppo uses these SQL snippets to compile a data pipeline that performs the specified aggregations and uses Eppo's stats engine to measure experiments' impact.
+
+At a minimum, a Fact SQL Definition needs to return a timestamp and a column to join to [assignment records](/data-management/definitions/assignment-sql) (for instance, `user_id`). That said, Fact SQL Definitions can point at anything from simple raw event logs to fully modeled aggregated data. One Fact SQL Definition can also contain multiple facts. For instance, a purchases table might contain a `gross_revenue` fact and a `net_revenue` fact.
+
+## Fact table schema
+
+Each Fact SQL Definition should return the following columns. Note that the specific names of these columns do not matter as you will map them into Eppo's data model when you create the fact definition.
+
+|           | Description| Examples |
+|-----------|------------|--------|
+| Timestamp | The time at which the event occurred, or the time dimension for aggregated data                            | `event_timestamp` , `session_timestamp` , `daydate` |
+| Entity ID | An ID for joining to assignment SQL sources. You can add multiple of these if the fact corresponds to multiple entities. | `user_id` , `device_id` , `company_id`|
+| Fact value (optional) | A numeric quantity associated with the fact. You can add several if an event has multiple associated values (e.g., net revenue and gross revenue) | `gross_revenue` , `items_added_to_cart` , `support_ticket_count` |
+| Fact property (optional) | A categorical variable associated with the event. This can be used to filter events in metric creation, or to break out experiment results. Note that properties that are 1:1 with experiment subjects should be defined in an [Assignment](/data-management/definitions/assignment-sql) or [Entity Property](/data-management/definitions/property-sql) SQL definition. | `revenue_category` , `support_ticket_reason` , `event_name` |
+| Partition date (optional) | An optional additional timestamp for filtering rows with a column other than the event timestamp. Useful if your event timestamp column differs from the table's partition timestamp column | `date` |
```

Comment:
Again, it would be good to link to the page on partitioning here: /data-management/warehouse-best-practices/#leveraging-partitioning